### PR TITLE
833 google sitemap

### PR DESCRIPTION
**Pull request by gpickin**
https://ortussolutions.atlassian.net/browse/CONTENTBOX-833

### CODE REVIEW

File: modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+* ---
+*/
+component{
+
+	// Module Properties
+	this.title 				= "ContentBox Sitemap Generator";
+	this.author 			= "Ortus Solutions, Corp";
+	this.webURL 			= "http://www.ortussolutions.com";
+	this.description 		= "Generates XML and HTML Sitemaps for your website";
+	this.version			= "1.0";
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc]: Can you change this to our standard: `this.version			= "3.1.0+@build.number@";`

File: modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+		};
+
+		// Layout Settings
+		layoutSettings = {
+			defaultLayout = ""
+		};
+
+		// datasources
+		datasources = {
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc]: remove this

File: modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+		};
+
+		// web services
+		webservices = {
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc]: remove this

File: modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	this.viewParentLookup 	= true;
+	// If true, looks for layouts in the parent first, if not found, then in module. Else vice-versa
+	this.layoutParentLookup = true;
+	// Module Entry Point
+	this.entryPoint			= "sitemap";
+
+	function configure(){
+
+		// parent settings
+		parentSettings = {
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc]: remove this

File: modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+		};
+
+		// module settings - stored in modules.name.settings
+		settings = {
+
+		};
+
+		// Layout Settings
+		layoutSettings = {
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/ModuleConfig.cfc]: remove this

File: modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+				search		= "",
+				isPublished	= true,
+				category	= "all",
+				author		= "all",
+				creator		= "all",
+				sortOrder	= "" 
+			); 		
+		}
+		 
+		if( rc.format == 'xml' ){
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc]: Guessing json is still missing?

File: modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+			return renderView( view="sitemap_xml", module="contentbox-sitemap" );
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc]: XML needs the xml content type set, else it looks like traditional request. I would change this to something like this:

```
event.renderData(
  data = renderView( view="sitemap_xml", module="contentbox-sitemap" ),
  contentType = "application/xml"
);
```

File: modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+		} else if( rc.format == 'txt' ){
+			return renderView( view="sitemap_txt", module="contentbox-sitemap" );
```
Comment (lmajano) [inline modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc]: Same here, leverage render data:


```
event.renderData(
  data = renderView( view="sitemap_txt", module="contentbox-sitemap" ),
  contentType = "text/plain"
);
```

File: modules/contentbox/models/system/CBHelper.cfc [hunk 9]
```diff
@@ -201,7 +201,7 @@ component accessors="true" singleton threadSafe{
 	function siteBaseURL(){
 		return replacenocase( 
 			getRequestContext().buildLink( linkTo='', ssl=getRequestContext().isSSL() ), 
-			"index.cfm", 
+			"index.cfm/",
```
Comment (lmajano) [inline modules/contentbox/models/system/CBHelper.cfc]: why add a ending slash?

File: modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc (hunk 6 above)
Comment (gpickin) [inline modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc]: I looked, but there is no JSON format online, unless we want it for another consumable entry point. Its not hard to add, but not recognized format for it.

File: modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc (hunk 7 above)
Comment (gpickin) [inline modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc]: Good point

File: modules/contentbox/models/system/CBHelper.cfc (hunk 9 above)
Comment (gpickin) [inline modules/contentbox/models/system/CBHelper.cfc]: Remember, its removing the `/`
By default it shows as `http://127.0.0.1:8080/index.cfm/`
So the base url replaces it and becomes `http://127.0.0.1//`

**Comment by gpickin** [comment]
Added JSON and added ContentType to renders

File: modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc (hunk 6 above)
Comment (gpickin) [inline modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc]: Added JSON

File: config/Routes.cfm [hunk 10]
```diff
@@ -29,18 +29,22 @@ limitations under the License.
...
+		rewriteURL = "index.cfm";
+	}
 	// Base URL
 	if( len(getSetting('AppMapping') ) lte 1){
-		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/index.cfm");
+		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/#rewriteURL#");
 	}
 	else{
-		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/#getSetting('AppMapping')#/index.cfm");
+		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/#getSetting('AppMapping')#/#rewriteURL#");
```
Comment (gpickin) [inline config/Routes.cfm]: What do you think about using this approach in the routing file for including or excluding the index.cfm?
This is a setting. 
We could set it using environment settings,

File: modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc (hunk 6 above)
Comment (gpickin) [inline modules/contentbox-ui/modules/contentbox-sitemap/handlers/main.cfc]: Added JSON

File: config/Routes.cfm (hunk 10 above)
Comment (lmajano) [inline config/Routes.cfm]: Well, the installer takes care of this on first run. So not sure if this is necessary.  Anyways, you could submit this as another pull so it does not pollute the sitemap one please.

**Comment by gpickin** [comment]
Ok, but how do we get this file updated for existing installs... we have to note it in the release notes, right?

**Comment by lmajano** [comment]
We can actually modify the `routes.cfm` in our patch process, so we can add that.  So just remove the rewrite piece and I can merge this in

**Comment by gpickin** [comment]
Oh ok, just a partial update? Thats cool. 
Removed the other pieces, review please.

**Pull request merged**